### ISSUE TITLE
Nginx with nginx-clojure module can't be compiled

### ISSUE THREAD

**Issue by glogo4j** [nginx-clojure/nginx-clojure#103]
My Linux version is ```2.6.32_1-14-0-0``` and glibc version is ```/lib/libc.so.6 -> libc-2.3.4.so```.
My Nginx with 1.4.7 version can't be compiled with nginx-clojure module, which mostly says like this:
```
In function `ngx_http_clojure_faccessat':
warning: implicit declaration of function `fstatat'
error: `AT_FDCWD' undeclared (first use in this function)
error: (Each undeclared identifier is reported only once
error: for each function it appears in.)
```
Dose the clojure-nginx compilation have some system limits?
Am I alone?

**Comment by xfeep** [comment]
@glogo4j  Thanks for your report!
`fstatat` was introduced since glibc 2.4 and Linux kernel 2.6.16. 
I have remove the use of it by commit 1080b28 because we do no need its additional functions comparing to `stat` .
Please again.

**Comment by glogo4j** [comment]
Thank you for your solution, it helps me fix last problem.
But unfortunatly, I got another problem.
```
make -f objs/Makefile
make[1]: Entering directory `/home/xxxx/nginx-pack/nginx-1.4.7'
cc -c -DNGX_CLOJURE_BE_SILENT_WITHOUT_JVM -I "/home/xxxx/.jumbo/opt/sun-java6/include" -I "/home/xxxx/.jumbo/opt/sun-java6/include/linux"  -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g   -I src/core -I src/event -I src/event/modules -I src/os/unix -I /home/xxxx/nginx-pack/pcre-8.36 -I objs -I src/http -I src/http/modules -I src/mail \
	-o objs/addon/c/ngx_http_clojure_shared_map_hashmap.o \
	/home/xxxx/nginx-pack/nginx-clojure/src/c/ngx_http_clojure_shared_map_hashmap.c
/home/xxxx/nginx-pack/nginx-clojure/src/c/ngx_http_clojure_shared_map_hashmap.c: In function `ngx_http_clojure_shared_map_hashmap_clear':
/home/xxxx/nginx-pack/nginx-clojure/src/c/ngx_http_clojure_shared_map_hashmap.c:533: error: structure has no member named `log_nomem'
/home/xxxx/nginx-pack/nginx-clojure/src/c/ngx_http_clojure_shared_map_hashmap.c:536: error: structure has no member named `last'
make[1]: *** [objs/addon/c/ngx_http_clojure_shared_map_hashmap.o] Error 1
make[1]: Leaving directory `/home/xxxx/nginx-pack/nginx-1.4.7'
make: *** [build] Error 2
```
xxxx means my home directory, and my jdk is 
```
java version "1.6.0_45"
Java(TM) SE Runtime Environment (build 1.6.0_45-b06)
Java HotSpot(TM) 64-Bit Server VM (build 20.45-b01, mixed mode)
```
It's a Sun JDK

**Comment by xfeep** [comment]
@glogo4j It has nothing to do with JDK version.It is caused by nginx version. In nginx 1.4.7 struct ngx_slab_pool_t has no member named log_nomem. I have fixed it by commit c467471 and tested it with nginx 1.4.7. Please try again.

**Comment by glogo4j** [comment]
Thank you so much!
Finally, I can compile it successfully.
Because of some reasons, I have to deal with "ancient" Nginx....
I will keep updating my experiences when I am using nginx-clojure.
Thanks again!
